projects/etherfi-lrt/index.js

```diff
@@ -7,13 +9,43 @@ const vaults = [
   '0xeDa663610638E6557c27e2f4e973D3393e844E70',
 ]
 
+const vaultAccountant = [
+  '0xbe16605B22a7faCEf247363312121670DFe5afBE',
+  '0x126af21dc55C300B7D0bBfC4F3898F558aE8156b',
+  '0x1b293DC39F94157fA0D1D36d7e0090C8B8B8c13F',
+  '0xBae19b38Bf727Be64AF0B578c34985c3D612e2Ba',
+  '0x1D4F0F05e50312d3E7B65659Ef7d06aa74651e0C',
+]
+
+async function vaultsTvl(api) {
+  for (let i = 0; i < vaults.length; i++) {
+    const bv = vaults[i]
+    const ba = vaultAccountant[i]
+    const bvSupply = await api.call({
+      target: bv,
+      abi: 'uint256:totalSupply',
```

Comment:
why call this on a different contract? why was previous code not working?

Reply:
there were multiple reason:
1) Some of the vaults don't directly hold the assets instead they own a contract that holds the assets so need to call totalSupply
2) some of the eth in weethk and weeths vaults were being doubled 
3) The totalSupply * quote -> value in the base asset we were not multipled by quote before (and the quote and totalSupply are from different contracts)